src/openforms/formio/dynamic_config/__init__.py

```diff
@@ -44,7 +45,25 @@ def rewrite_formio_components_for_request(
     Loop over the formio configuration and inject request-specific configuration.
     """
     for component in configuration_wrapper:
-        register.update_config_for_request(component, request=request)
+        # prevent multiple overwrites of content components (bandaid for #2895)
+        # postprocess (a) non-content components, (b) components with inline style
+        # (c) components with nonce that does not match the nonce on the request
+
+        # get HTML (if any) from component and parse it
+        if html_string := component.get("html", None):
+            request_nonce = request.headers.get("X-Csp-Nonce", None)
+            html = lxml.html.fromstring(html_string)
+            tags = list(html.iter())
+            tags_with_inline_style = [tag for tag in tags if tag.get("style")]
+            tags_with_nonce = [tag for tag in tags if tag.get("nonce")]
```

Comment:
shouldn't this be limited to the `tags_with_inline_style` instead of all `tags`?